tests/app/questionnaire/test_path_finder.py

```diff
@@ -1,720 +1,632 @@
+        section_id="default-section",
+    )
+
+    answer_1 = Answer(answer_id="mandatory-checkbox-answer", value="Cheese")
+    answer_2 = Answer(answer_id="non-mandatory-checkbox-answer", value="deep pan")
+    answer_3 = Answer(answer_id="single-checkbox-answer", value="Estimate")
+
+    answer_store.add_or_update(answer_1)
+    answer_store.add_or_update(answer_2)
+    answer_store.add_or_update(answer_3)
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "default-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["mandatory-checkbox"],
+            }
+        ]
+    )
+
+    path_finder = PathFinder(schema, answer_store, list_store, progress_store, {}, {})
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    assert routing_path == expected_path
+
+
+def test_routing_path_with_conditional_value_not_in_metadata(answer_store, list_store):
+    schema = load_schema_from_name("test_metadata_routing")
+    section_id = schema.get_section_id_for_block_id("block1")
+    expected_path = RoutingPath(
+        ["block1", "block2", "block3"], section_id="default-section"
+    )
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "default-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["block1"],
+            }
+        ]
+    )
+
+    path_finder = PathFinder(schema, answer_store, list_store, progress_store, {}, {})
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    assert routing_path == expected_path
+
+
+def test_routing_path_should_skip_block(answer_store, list_store):
+    # Given
+    schema = load_schema_from_name("test_skip_condition_block")
+    section_id = schema.get_section_id_for_block_id("should-skip")
+    answer_store.add_or_update(
+        Answer(answer_id="do-you-want-to-skip-answer", value="Yes")
+    )
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "introduction-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["do-you-want-to-skip"],
+            }
+        ]
+    )
+
+    # When
+    path_finder = PathFinder(
+        schema,
+        answer_store,
+        list_store,
+        progress_store,
+        {},
+        {},
+    )
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    # Then
+    expected_routing_path = RoutingPath(
+        ["do-you-want-to-skip", "a-non-skipped-block"],
+        section_id="default-section",
+    )
+
+    assert routing_path == expected_routing_path
+
+
+def test_new_routing_path_should_skip_block(answer_store, list_store):
+    # Given
+    schema = load_schema_from_name("test_new_skip_condition_block")
```

Comment:
Aren't new and old skipping/routing the same bar the schema? If so we should parametrize them

Reply:
We have tests for both the new and old rules in some instances i.e test_new_routing_path_should_skip_group and test_routing_path_should_skip_group. They look the same to me but the schema, which we could parametrize

Reply:
This applies to:
1. test_new_routing_path_with_conditional_path / test_routing_path_with_conditional_path
1. test_new_routing_path_should_skip_block / test_routing_path_should_skip_block
2. test_new_routing_path_should_skip_group / test_routing_path_should_skip_group
3. test_new_remove_answer_and_block_if_routing_backwards / test_remove_answer_and_block_if_routing_backwards


For 1, `test_routing_path_with_conditional_path` is actually pointing at a new schema instead of old.
For 2, it is almost identical but schema differ slightly, but can still parametrize by passing the expected path block ids also.
For 3, schema behaves slightly different, so would be temped to leave as is.